contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3660,3 +3660,91 @@ END
+        	CAST (CASE v.type_name
+			WHEN 'int' THEN 8
+                        WHEN 'nchar' THEN 10
+                        WHEN 'char' THEN 3
+                        WHEN 'date' THEN 31
+                        WHEN 'nvarchar' THEN 12
+                        WHEN 'varchar' THEN 22
+                        WHEN 'table' THEN 23
+                        WHEN 'datetime' THEN 4
+                        WHEN 'datetime2' THEN 33
+                        WHEN 'datetimeoffset' THEN 34
+                        WHEN 'smalldatetime' THEN 15
+			WHEN 'time' THEN 32
+                        WHEN 'decimal' THEN 5
+			WHEN 'numeric' THEN 5
+                        WHEN 'float' THEN 6
+                        WHEN 'real' THEN 13
+                        WHEN 'nchar' THEN 10
+                        WHEN 'flag' THEN 2
+                        WHEN 'money' THEN 9
+                        WHEN 'smallmoney' THEN 17
+                        WHEN 'tinyint' THEN 20
+                        WHEN 'smallint' THEN 16
+                        WHEN 'bigint' THEN 0
+                        WHEN 'bit' THEN 2
+			WHEN 'text' THEN 18
+			WHEN 'ntext' THEN 11
+			WHEN 'binary' THEN 1
+			WHEN 'varbinary' THEN 21
+			WHEN 'image' THEN 7
+                        ELSE 0 END
+                	AS smallint) AS [MANAGED_DATA_TYPE],
+        	CAST (CASE 
+			WHEN v.type_name IN (N'nchar', N'nvarchar') AND p.max_length <> -1 THEN p.max_length / 2
+			WHEN v.type_name IN (N'char', N'varchar', N'binary', N'varbinary') AND p.max_length <> -1 THEN p.max_length
+			WHEN v.type_name IN (N'nvarchar', N'varchar', N'varbinary') AND p.max_length = -1 THEN 0
+                	WHEN v.type_name IN (N'text', N'image') THEN 2147483647
+                	WHEN v.type_name = 'ntext' THEN 1073741823
+                	ELSE NULL END 
+			AS INT) AS [CHARACTER_MAXIMUM_LENGTH],
+        	CAST(CASE 
+			WHEN v.type_name IN (N'int', N'smallint', N'bigint', N'tinyint', N'float', N'real', N'decimal', N'numeric', N'money', N'smallmoney') 
+				THEN v.PRECISION
+			ELSE NULL END 
+			AS smallint) AS [NUMERIC_PRECISION],
+        	CAST(CASE 
+			WHEN v.type_name IN (N'decimal', N'numeric') THEN v.SCALE 
+			ELSE NULL END 
+			AS smallint ) AS [NUMERIC_SCALE],
+        	CAST(NULL AS sys.nvarchar(128)) AS [TYPE_CATALOG_NAME],
+        	CAST(NULL AS sys.nvarchar(128)) AS [TYPE_SCHEMA_NAME],
+        	CAST(v.TYPE_NAME AS sys.nvarchar(128)) AS [TYPE_NAME],
+        	CAST(NULL AS sys.nvarchar(128)) AS XML_CATALOGNAME,
+        	CAST(NULL AS sys.nvarchar(128)) AS XML_SCHEMANAME,
+        	CAST(NULL AS sys.nvarchar(128)) AS XML_SCHEMACOLLECTIONNAME,
+        	CAST(CASE
+			WHEN v.type_name = 'datetime' THEN 3
+                    	WHEN v.type_name IN (N'datetime2', N'datetimeoffset', N'time') THEN 7
+			WHEN v.type_name IN (N'date', N'smalldatetime') THEN 0
+                    	ELSE NULL END AS int) AS [SS_DATETIME_PRECISION]
+   	FROM sys.sp_sproc_columns_view v
+   	LEFT OUTER JOIN sys.all_parameters AS p 
```

Comment:
Have we looked into solution where we don't need to have join with `sys.all_parameters` instead we calculate it, If we just need `max_length` column from it?

Reply:
Yes, I have looked in that direction. But, I don't think we can do that.

Reply:
I'm still not sure why we are doing in this way `sys.sp_sproc_columns_view` view itself contains `LENGTH` columns or we can think of using join with sys.types. Lets discuss this offline.